packages/server/src/insights/inisghts.controller.ts

```diff
@@ -0,0 +1,52 @@
+import { InsightsService } from './insights.service';
+import {
+  Controller,
+  UseGuards,
+  UseInterceptors,
+  ClassSerializerInterceptor,
+  Get,
+  Param,
+  Query,
+} from '@nestjs/common';
+import { JwtAuthGuard } from 'login/jwt-auth.guard';
+import { Connection } from 'typeorm';
+import { Roles } from 'profile/roles.decorator';
+import { Role, GetInsightResponse, ERROR_MESSAGES } from '@koh/common';
+import { INSIGHTS } from './insights';
+
+@Controller('insights')
+@UseGuards(JwtAuthGuard)
+@UseInterceptors(ClassSerializerInterceptor)
+export class InsightsController {
+  constructor(
+    private connection: Connection,
+    private insightsService: InsightsService,
+  ) {}
+
+  @Get(':courseId/:insightName')
+  @Roles(Role.PROFESSOR)
+  async get(
+    @Param('courseId') courseId: number,
+    @Param('insightName') insightName: string,
+    @Query() filters: any,
+  ): Promise<GetInsightResponse> {
+    // // TODO: Check that the current role is allowed to see the given insight
```

Comment:
is this needed if the whole endpoint is guarded by role Professor?

Reply:
We definitely need one or the other. To make things extensible so that in the future we can allow certain insights to be shown to TA's as well, I'll remove the Professor role guard and change the code so the insight access up to the `roles` attribute of the insight itself.